gnucash/report/commodity-utilities.scm

```diff
@@ -764,6 +765,18 @@
          (gnc:exchange-by-pricedb-nearest
           foreign domestic to-date-tp))))))
 
+(define (gnc:case-price-fn source target-curr date)
+  (define pdb (gnc-pricedb-get-db (gnc-get-current-book)))
+  (case source
+    ((pricedb-nearest) (cut gnc-pricedb-get-price-at-date pdb <> target-curr date))
+    ((pricedb-latest)  (cut gnc-pricedb-get-price-latest pdb <> target-curr))
+    (else
+     (lambda (commodity)
+       (let* ((exchange-fn (gnc:case-exchange-fn source target-curr date))
+              (foreign-mon (gnc:make-gnc-monetary commodity 1))
```

Comment:
@christopherlam You might put back @cstim's old hack where you convert 1000 instead of 1. Then you can div that by 1000 and have 6 decimals for your price.

Reply:
I don't think this is needed. I've confirmed that: for all 4 price sources, and force-decimal true/false, the above code works well. Report currency USD:
£1.00	$1.8570
JP¥1	$0.0096

£1.00	$1.3484
JP¥1	$0.0096

£1.00	$1 + 24967/71671
JP¥1	$1/104

£1.00	$1 + 1157/1350
JP¥1	$1/104

For report currency JPY:

nearest in time, force-decimal off
£1.00	JP¥138.00
$1.00	JP¥104.00

average-cost, force-decimal off
£1.00	JP¥148 + 76956/128329
$1.00	JP¥104.00

average-cost, force-decimal on
£1.00	JP¥148.60
$1.00	JP¥104.00

nearest-in-time, force-decimal on
£1.00	JP¥138.00
$1.00	JP¥104.00

Reply:
OK, the one of interest is average-cost. I'm a bit surprised that it doesn't round to JPY's SCU (1) given your earlier complaint and that AFAICT you haven't changed anything related to average-cost or weighted-average.

Reply:
I think this JPY issuing 2 decimal points is a completely separate issue; I'd merge this branch first then investigate JPY separately.